parsl/monitoring/visualization/plots/default/workflow_resource_plots.py

```diff
@@ -91,3 +91,67 @@ def resource_time_series(tasks, type='psutil_process_time_user', label='CPU user
                              yaxis=dict(title=label),
                              title=label))
     return plot(fig, show_link=False, output_type="div", include_plotlyjs=False)
+
+
+def resource_efficiency(resource, node, label='CPU'):
+    try:
+        resource['epoch_time'] = (pd.to_datetime(
+            resource['timestamp']) - pd.Timestamp("1970-01-01")) // pd.Timedelta('1s')
+        node['epoch_time'] = (pd.to_datetime(
+            node['reg_time']) - pd.Timestamp("1970-01-01")) // pd.Timedelta('1s')
+        resource = resource.sort_values(by='epoch_time')
+        start = min(resource['epoch_time'].min(), node['epoch_time'].min())
+        end = resource['epoch_time'].max()
+        resource['relative_time'] = resource['epoch_time'] - start
+        node['relative_time'] = node['epoch_time'] - start
+
+        task_plot = [0] * (end - start + 1)
+        if label == 'CPU':
+            total = node['cpu_count'].sum()
+        elif label == 'mem':
+            total = node['total_memory'].sum() / 1024 / 1024 / 1024
+
+        for task_id in resource['task_id'].unique():
+            tmp = resource[resource['task_id'] == task_id]
+            tmp['last_timestamp'] = tmp['relative_time'].shift(1)
+            if label == 'CPU':
+                tmp['last_cputime'] = tmp['psutil_process_time_user'].shift(1)
+            for index, row in tmp.iterrows():
+                if np.isnan(row['last_timestamp']):
+                    continue
+                for i in range(int(row['last_timestamp']), int(row['relative_time'])):
+                    if label == 'CPU':
+                        diff = (row['psutil_process_time_user'] - row['last_cputime']) / (row['relative_time'] - row['last_timestamp'])
+                    elif label == 'mem':
+                        diff = row['psutil_process_memory_resident'] / 1024 / 1024 / 1024
+                    task_plot[i] += diff
+
+        if label == 'CPU':
+            name1 = 'Used CPU cores'
+            name2 = 'Requested CPU cores'
```

Comment:
I dislike the phrase `requested` here and in the memory usage. In both cases, it is the total amount of CPU or memory being reported by the systems - not the amount requested.